[Integration Tests] note-export routes test returns 410 — pg_cron expiry job runs during CI and expires test data

## Problem

`tests/note-export/routes.test.ts` intermittently returns 410 (Gone/Expired) instead of 200 during CI integration test runs.

## Current Behavior

Tests fail with:
```
AssertionError: expected 410 to be 200
Response: { error: 'Export has expired' }
```

## Expected Behavior

Tests should get 200 with the export data.

## Root Cause

The `note_export` table has a pg_cron expiry job (installed by migration 166) that marks records with `expires_at < NOW()` as expired. When the CI integration test run is long (~30+ minutes), the pg_cron job fires during the run and expires test data before the `note-export/routes.test.ts` tests execute.

## Impact

Intermittent CI failures on any PR. Blocks merging.

## Fix Options

1. Test records should be created with a far-future `expires_at` (e.g. NOW() + interval '1 year')
2. OR: pg_cron expiry job should be disabled during test runs
3. OR: `beforeEach` in the test should refresh test data

## Context

- First observed: 2026-03-14 on PR #2580
- Previous CI runs on same main-branch code passed
- Affects: `tests/note-export/routes.test.ts` — 4 tests failing

## Related

- Migration 166: `166_note_export_schema.up.sql` (installs the pg_cron job)
- Epic: #2475

--- a/tests/note-export/routes.test.ts
+++ b/tests/note-export/routes.test.ts
@@ -32,6 +32,14 @@ const mockCreateExportJob = vi.mocked(createExportJob);
 const mockGetExportById = vi.mocked(getExportById);
 const mockRunExportJob = vi.mocked(runExportJob);
 
+// Use a far-future expires_at so that the route's inline expiry check
+// (exportRow.expires_at < new Date()) never triggers during tests, regardless
+// of how long the CI run takes or when pg_cron fires.  The assertion in the
+// "formats dates as ISO strings" test below must match this constant.
+const TEST_EXPIRES_AT = new Date('2099-12-31T12:00:00Z');
+const TEST_CREATED_AT = new Date('2026-03-13T12:00:00Z');
+const TEST_UPDATED_AT = new Date('2026-03-13T12:00:00Z');
+
 function makeExportRow(overrides: Record<string, unknown> = {}) {
   return {
     id: EXPORT_UUID,
@@ -48,9 +56,9 @@ function makeExportRow(overrides: Record<string, unknown> = {}) {
     size_bytes: null,
     attempt_count: 0,
     started_at: null,
-    expires_at: new Date('2026-03-14T12:00:00Z'),
-    created_at: new Date('2026-03-13T12:00:00Z'),
-    updated_at: new Date('2026-03-13T12:00:00Z'),
+    expires_at: TEST_EXPIRES_AT,
+    created_at: TEST_CREATED_AT,
+    updated_at: TEST_UPDATED_AT,
   } as Record<string, unknown>;
   // Return as Record to allow overrides to set any property
 }
@@ -71,9 +79,9 @@ function makeTypedExportRow(overrides: Record<string, unknown> = {}) {
     size_bytes: null,
     attempt_count: 0,
     started_at: null,
-    expires_at: new Date('2026-03-14T12:00:00Z'),
-    created_at: new Date('2026-03-13T12:00:00Z'),
-    updated_at: new Date('2026-03-13T12:00:00Z'),
+    expires_at: TEST_EXPIRES_AT,
+    created_at: TEST_CREATED_AT,
+    updated_at: TEST_UPDATED_AT,
     ...overrides,
   };
   return base;
@@ -475,8 +483,8 @@ describe('Export Routes (Issue #2478)', () => {
     it('returns exports after direct DB insert', async () => {
       const { app } = buildTestApp();
       await pool.query(
-        `INSERT INTO note_export (id, namespace, requested_by, source_type, source_id, format)
-         VALUES ($1, $2, $3, 'note', $4, 'pdf')`,
+        `INSERT INTO note_export (id, namespace, requested_by, source_type, source_id, format, expires_at)
+         VALUES ($1, $2, $3, 'note', $4, 'pdf', NOW() + INTERVAL '1 year')`,
         [EXPORT_UUID, TEST_NS, TEST_USER, NOTE_UUID],
       );
 
@@ -495,13 +503,13 @@ describe('Export Routes (Issue #2478)', () => {
     it('filters by status', async () => {
       const { app } = buildTestApp();
       await pool.query(
-        `INSERT INTO note_export (namespace, requested_by, source_type, source_id, format, status)
-         VALUES ($1, $2, 'note', $3, 'pdf', 'pending')`,
+        `INSERT INTO note_export (namespace, requested_by, source_type, source_id, format, status, expires_at)
+         VALUES ($1, $2, 'note', $3, 'pdf', 'pending', NOW() + INTERVAL '1 year')`,
         [TEST_NS, TEST_USER, NOTE_UUID],
       );
       await pool.query(
-        `INSERT INTO note_export (namespace, requested_by, source_type, source_id, format, status, storage_key)
-         VALUES ($1, $2, 'note', $3, 'pdf', 'ready', 'exports/test.pdf')`,
+        `INSERT INTO note_export (namespace, requested_by, source_type, source_id, format, status, storage_key, expires_at)
+         VALUES ($1, $2, 'note', $3, 'pdf', 'ready', 'exports/test.pdf', NOW() + INTERVAL '1 year')`,
         [TEST_NS, TEST_USER, NOTE_UUID],
       );
 
@@ -521,8 +529,8 @@ describe('Export Routes (Issue #2478)', () => {
       const { app } = buildTestApp();
       for (let i = 0; i < 3; i++) {
         await pool.query(
-          `INSERT INTO note_export (namespace, requested_by, source_type, source_id, format)
-           VALUES ($1, $2, 'note', $3, 'pdf')`,
+          `INSERT INTO note_export (namespace, requested_by, source_type, source_id, format, expires_at)
+           VALUES ($1, $2, 'note', $3, 'pdf', NOW() + INTERVAL '1 year')`,
           [TEST_NS, TEST_USER, NOTE_UUID],
         );
       }
@@ -544,8 +552,8 @@ describe('Export Routes (Issue #2478)', () => {
     it('does not show other users exports', async () => {
       const { app } = buildTestApp();
       await pool.query(
-        `INSERT INTO note_export (namespace, requested_by, source_type, source_id, format)
-         VALUES ($1, 'other@example.com', 'note', $2, 'pdf')`,
+        `INSERT INTO note_export (namespace, requested_by, source_type, source_id, format, expires_at)
+         VALUES ($1, 'other@example.com', 'note', $2, 'pdf', NOW() + INTERVAL '1 year')`,
         [TEST_NS, NOTE_UUID],
       );
 
@@ -647,7 +655,7 @@ describe('Export Routes (Issue #2478)', () => {
 
       expect(res.statusCode).toBe(200);
       const body = res.json();
-      expect(body.expires_at).toBe('2026-03-14T12:00:00.000Z');
+      expect(body.expires_at).toBe('2099-12-31T12:00:00.000Z');
       expect(body.created_at).toBe('2026-03-13T12:00:00.000Z');
       await app.close();
     });

